libobs-opengl/gl-egl-common.c

```diff
@@ -215,6 +217,125 @@ gl_egl_create_dmabuf_image(EGLDisplay egl_display, unsigned int width,
 	return texture;
 }
 
+EGLint query_dmabuf_formats(EGLDisplay egl_display, EGLint **formats)
+{
+	EGLint max_formats = 0;
+	EGLint *format_list = NULL;
+
+	if (!glad_eglQueryDmaBufFormatsEXT) {
+		blog(LOG_ERROR, "Unable to load eglQueryDmaBufFormatsEXT");
+		return 0;
+	}
+
+	if (!glad_eglQueryDmaBufFormatsEXT(egl_display, 0, NULL,
+					   &max_formats)) {
+		blog(LOG_ERROR, "Cannot query the number of formats: %s",
+		     gl_egl_error_to_string(eglGetError()));
+		return 0;
+	}
+
+	format_list = bzalloc(max_formats * sizeof(EGLint));
+	if (!format_list) {
+		blog(LOG_ERROR, "Unable to allocate memory");
+		return 0;
+	}
+
+	if (!glad_eglQueryDmaBufFormatsEXT(egl_display, max_formats,
+					   format_list, &max_formats)) {
+		blog(LOG_ERROR, "Cannot query a list of formats: %s",
+		     gl_egl_error_to_string(eglGetError()));
+		free(format_list);
+		return 0;
+	}
+
+	*formats = format_list;
+	return max_formats;
+}
+
+EGLint query_dmabuf_modifiers(EGLDisplay egl_display, EGLint drm_format,
+			      EGLuint64KHR **modifiers)
+{
+	EGLint max_modifiers = 0;
+	EGLuint64KHR *modifier_list = NULL;
+	EGLBoolean *external_only = NULL;
+
+	if (!glad_eglQueryDmaBufModifiersEXT) {
+		blog(LOG_ERROR, "Unable to load eglQueryDmaBufModifiersEXT");
+		return 0;
+	}
+
+	if (!glad_eglQueryDmaBufModifiersEXT(egl_display, drm_format, 0, NULL,
+					     NULL, &max_modifiers)) {
+		blog(LOG_ERROR, "Cannot query the number of modifiers: %s",
+		     gl_egl_error_to_string(eglGetError()));
+		return 0;
+	}
+
+	modifier_list = bzalloc(max_modifiers * sizeof(EGLuint64KHR));
+	external_only = bzalloc(max_modifiers * sizeof(EGLBoolean));
```

Comment:
external_only is not used in this code and optional in the query function so lets just remove it.

Reply:
it is used in `glad_eglQueryDmaBufModifiersEXT`.

Reply:
It is option in `glad_eglQueryDmaBufModifiersEXT` at least according to the spec, are you saying that is not the case?

Reply:
Sry I don't get what you want to say me. What i wanted to say is:
`external_only` is an argument for `glad_eglQueryDmaBufModifiersEXT` and i don't know if according to the spec I can set this pointer just to `NULL` and it will skip it or not. So I just allocate the memory for it, provide it to the egl call and free it afterwards.

Reply:
I mean, we should pass NULL here unless you are saying it crashes with that argument.

Reply:
Works using NULL